src/net/server/middleware/tsig.rs

```diff
@@ -0,0 +1,506 @@
+//! RFC 8495 TSIG message authentication middleware.
+//!
+//! This module provides a TSIG request validation and response signing
+//! middleware service. The underlying TSIG RR processing is implemented using
+//! the [`rdata::tsig`][crate::rdata::tsig] module.
+//!
+//! Signed requests thta fail signature verification will be rejected.
+//!
+//! Unsigned requests and correctly signed requests will pass through this
+//! middleware unchanged.
+//!
+//! For requests which were correctly signed the corresponding response(s)
+//! will be signed using the same key as the request.
+//!
+//! # Determining the key that a request was signed with
+//!
+//! The key that signed a request is output by this middleware via the request
+//! metadata in the form [`Option<KS::Key>`], where `KS` denotes the type of
+//! [`KeyStore`] that was used to construct this middleware. Upstream services
+//! can choose to ignore the metadata by being generic over any kind of
+//! metadata, or may offer a [`Service`] impl that specifically accepts the
+//! [`Option<KS::Key>`] metadata type, enabling the upstream service to use
```

Comment:
```suggestion
//! `Option<KS::Key>` metadata type, enabling the upstream service to use
```
It's not a valid link because of `KS`